test/unit/services/test_image.py

```diff
@@ -139,19 +144,23 @@ def check_create_success(self) -> None:
 
         assert self._created_image == self._expected_image
 
-    def check_create_failed_with_exception(self, message: str, assert_checks: bool = True) -> None:
+    def check_create_failed_with_exception(self, message: str, assert_count: bool = True) -> None:
         """
         Checks that a prior call to `call_create_expecting_error` worked as expected, raising an exception
         with the correct message.
 
         :param message: Message of the raised exception.
+        :param assert_count: Whether the `count_by_entity_id` repo method is expected to be called or not.
         """
-        if assert_checks:
-            self.mock_generate_thumbnail_base64_str.assert_called_once_with(self._upload_file)
-            self.mock_image_store.upload.assert_called_once_with(
-                str(self._expected_image_id), self._image_post_metadata, self._upload_file
+        if assert_count:
+            self.mock_image_repository.count_by_entity_id.assert_called_once_with(
+                str(self._expected_image_in.entity_id)
             )
-        self.mock_image_repository.create.assert_not_called()
+        else:
+            self.mock_image_repository.count_by_entity_id.create.assert_not_called()
+            self.mock_generate_thumbnail_base64_str.create.assert_not_called()
+            self.mock_image_store.upload.create.assert_not_called()
+            self.mock_image_repository.create.assert_not_called()
 
```

Comment:
```suggestion
        else:
            self.mock_image_repository.count_by_entity_id.create.assert_not_called()

        self.mock_generate_thumbnail_base64_str.create.assert_not_called()
        self.mock_image_store.upload.create.assert_not_called()
        self.mock_image_repository.create.assert_not_called()

```
should these be in the else block?